crates/ra_mbe/src/syntax_bridge.rs

```diff
@@ -93,8 +117,14 @@ fn convert_tt(
                 }
             }
             SyntaxElement::Node(node) => {
-                let child = convert_tt(token_map, global_offset, node)?.into();
-                token_trees.push(child);
+                let mut child = convert_tt(token_map, global_offset, node, allow_no_delimiter)?;
+
+                if child.delimiter == tt::Delimiter::None {
+                    // Flatten token tree
```

Comment:
I am not sure if we should be flattening the trees. The None delimiter is useful to avoid flattening. For example, if you catch `1+2` as a meta var `$`, and paste it into `$e * 3`, the tokens would be `_1 + 2_ * 3`, where `_ _` are the fake-parenthesis -- none delimiters.